cudax/include/cuda/experimental/__stf/internal/logical_data.cuh

```diff
@@ -2279,26 +2280,41 @@ public:
   {
     using U = readonly_type_of<T>;
     // The constness of *this implies that access mode is read
-    return task_dep<U>(*this, /* access_mode::read, */ ::std::forward<Pack>(pack)...);
+    return task_dep<U, ::std::monostate, false>(*this, /* access_mode::read, */ ::std::forward<Pack>(pack)...);
```

Comment:
Critical: I am wondering why `access_mode::read` is commented out here, while all the others take the argument

Reply:
It's true that it's surprising !

This later allow to dispatch between either constructors, the second one does not take an access_mode and corresponds to read_only access mode and has an extra const_cast.

```
 // dependency with an explicit data_place
  task_dep_untyped(const logical_data_untyped& d,
                   access_mode m,
                   data_place dplace,
                   ::std::shared_ptr<reduction_operator_base> redux_op = nullptr)
      : data(pack_state(d))
      , m(m)
      , dplace(mv(dplace))
      , redux_op(mv(redux_op))
  {}

  task_dep_untyped(
    const logical_data_untyped& d, data_place dplace, ::std::shared_ptr<reduction_operator_base> redux_op = nullptr)
      : task_dep_untyped(const_cast<logical_data_untyped&>(d), access_mode::read, mv(dplace), mv(redux_op))
  {}
```
It's not super elegant ...

Reply:
I've added comments to explain what is going on under the hood. This is something that should be improved in future work by adding tag types to do this dispatching in a better way.